cachito/workers/tasks/gomod.py

```diff
@@ -16,45 +17,96 @@
+            log.warning("go.mod file%s missing for request at %s", file_suffix, invalid_files_print)
             return
 
-        raise CachitoError("The go.mod file is missing")
-
-    log.info("Fetching gomod dependencies for request %d", request_id)
-    request = set_request_state(request_id, "in_progress", "Fetching the gomod dependencies")
-    try:
-        gomod = resolve_gomod(str(bundle_dir.source_dir), request, dep_replacements)
-    except CachitoError:
-        log.exception("Failed to fetch gomod dependencies for request %d", request_id)
-        raise
-
-    env_vars = {
-        "GOCACHE": {"value": "deps/gomod", "kind": "path"},
-        "GOPATH": {"value": "deps/gomod", "kind": "path"},
-    }
-    env_vars.update(config.cachito_default_environment_variables.get("gomod", {}))
-    update_request_with_package(request_id, gomod["module"], env_vars)
-    update_request_with_deps(request_id, gomod["module"], gomod["module_deps"])
-
-    # add package deps
-    for package in gomod["packages"]:
-        if package.get("pkg_deps"):
-            # This also adds the package to the request
-            update_request_with_deps(request_id, package["pkg"], package["pkg_deps"])
+        raise CachitoError(
+            "The {} file{} must be present for the gomod package manager".format(
+                invalid_files_print.strip(), file_suffix
+            )
+        )
+
+    for i, subpath in enumerate(subpaths):
+        log.info(
+            "Fetching the gomod dependencies for request %d in subpath %s", request_id, subpath
+        )
+        request = set_request_state(
+            request_id,
+            "in_progress",
+            f'Fetching the gomod dependencies at the "{subpath}" directory',
+        )
+        gomod_source_path = str(bundle_dir.app_subpath(subpath).source_dir)
+        try:
+            gomod = resolve_gomod(gomod_source_path, request, dep_replacements)
+        except CachitoError:
+            log.exception("Failed to fetch gomod dependencies for request %d", request_id)
+            raise
+
+        if i == 0:
+            env_vars = {
+                "GOCACHE": {"value": "deps/gomod", "kind": "path"},
+                "GOPATH": {"value": "deps/gomod", "kind": "path"},
+            }
+            env_vars.update(config.cachito_default_environment_variables.get("gomod", {}))
         else:
-            update_request_with_package(request_id, package["pkg"])
+            env_vars = None
+        update_request_with_package(request_id, gomod["module"], env_vars)
+        update_request_with_deps(request_id, gomod["module"], gomod["module_deps"])
+
+        # add package deps
+        for package in gomod["packages"]:
+            if package.get("pkg_deps"):
+                # This also adds the package to the request
+                update_request_with_deps(request_id, package["pkg"], package["pkg_deps"])
+            else:
+                update_request_with_package(request_id, package["pkg"])
```

Comment:
Question for future self: do these packages have the same subpath as the module? If not, how does one find the subpath?